packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -729,4 +1037,307 @@ describe("ControllerHelper: mainnet fork", function () {
+         // lpWPowerPerpPool: [recipient, vaultId, wPowerPerpAmount, collateralToDeposit, collateralToLP, amount0Min, amount1Min, lowerTick, upperTick ]
+         data: abiCoder.encode(["address", "uint256", 'uint256','uint256', 'uint256', 'uint256', 'uint256', 'int24', 'int24'],
+          [controllerHelper.address, vaultId, BigNumber.from(0), BigNumber.from(0), wethAmountToLP, amount0Min, amount1Min, newTickLower, newTickUpper])
+       }
+    ]
+    const flashLoanAmount = ethers.utils.parseUnits('50')
+    await controllerHelper.connect(depositor).rebalanceVaultNft(vaultId, flashLoanAmount, params);
+    const depositorSqueethBalanceAfter = await wSqueeth.balanceOf(depositor.address)
+    const depositorEthBalanceAfter = await ethers.provider.getBalance(depositor.address)
+
+    // Get new vault and LP info
+    const vaultAfter = await controller.vaults(vaultId);
+    const tokenIdAfter = vaultAfter.NftCollateralId;
+    const positionAfter = await (positionManager as INonfungiblePositionManager).positions(tokenIdAfter)
+    const [amount0After, amount1After] = await (positionManager as INonfungiblePositionManager).connect(controllerSigner).callStatic.decreaseLiquidity({
+      tokenId: tokenIdAfter,
+      liquidity: positionAfter.liquidity,
+      amount0Min: 0,
+      amount1Min: 0,
+      deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+    })
+    const wPowerPerpAmountInLPAfter = (isWethToken0) ? amount1After : amount0After;
+    const wethAmountInLPAfter = (isWethToken0) ? amount0After : amount1After;
+    // Changes
+    const depositorEthdiff = depositorEthBalanceAfter.sub(depositorEthBalanceBefore)
+    const depositorSqueethdiff = depositorSqueethBalanceAfter.sub(depositorSqueethBalanceBefore)
+    const vaultEthDiff = vaultAfter.collateralAmount.sub(vaultBefore.collateralAmount)
+    const vaultSqueethDiff = vaultAfter.shortAmount.sub(vaultBefore.shortAmount)
+    const lpEthDiff = wethAmountInLPAfter.sub(wethAmountInLPBefore)
+    const lpSqueethDiff = wPowerPerpAmountInLPAfter.sub(wPowerPerpAmountInLPBefore)
+    // Assertions
+    console.log('currentTick', currentTick)
+    console.log('positionAfter.TickLower', positionAfter.tickLower)
+    console.log('positionAfter.TickUpper', positionAfter.tickUpper)
+    console.log('wPowerPerpAmountInLPAfter', wPowerPerpAmountInLPAfter.toString())
+    console.log('wethAmountInLPAfter', wethAmountInLPAfter.toString())
+    console.log('lpSqueethDiff', lpSqueethDiff.toString())
+    console.log('lpEthDiff', lpEthDiff.toString())
+
+    expect(positionAfter.tickLower === newTickLower).to.be.true
+    expect(positionAfter.tickUpper === newTickUpper).to.be.true
+    expect(vaultAfter.shortAmount.eq(vaultBefore.shortAmount)).to.be.true
+    expect(vaultAfter.collateralAmount.eq(vaultBefore.collateralAmount)).to.be.true
+    expect(vaultAfter.NftCollateralId==vaultBefore.NftCollateralId).to.be.false
+    expect(wPowerPerpAmountInLPAfter.eq(BigNumber.from(0))).to.be.true
```

Comment:
maybe should check that eth in LP is = eth before + chg in depositor eth balance (conservation of eth!)

Reply:
yes indeed - I had but removed because I can't get the tx for decreaseLiquidity static call so it can't get it to rec exactly. Could just check before and after balances but very hacky. Is there a better way?